lib/middleware.py

```diff
@@ -1,19 +1,35 @@
 from django.conf import settings
-from django.http import HttpResponseRedirect
+from django.http import Http404, HttpResponseRedirect
 from django.utils import translation
 from django.utils.cache import patch_vary_headers
 from django.utils.deprecation import MiddlewareMixin
+from django.shortcuts import redirect
+
+from urllib.parse import urlparse
 
 
 class LocaleMiddleware(MiddlewareMixin):
 
     response_redirect_class = HttpResponseRedirect
 
     def process_request(self, request):
-        if hasattr(request, 'user') and request.user.is_authenticated:
+        language = translation.get_language_from_request(request)
+        current_url = request.META.get('PATH_INFO')
+        last_url = request.META.get('HTTP_REFERER', '')
```

Comment:
Yeah, this doesn't look very nice way to do this. I'll return on this after thinking a bit of ideas where to look a better solution.